library/bigip_user.py

```diff
@@ -182,411 +172,366 @@
-        else:
-            return False
-        params['name'] = self.params['username_credential']
-        params['partition'] = self.params['partition']
-        self.update_user_on_device(params)
-        return True
+    def validate_create_parameters(self):
+        """Password credentials and partition access are mandatory,
 
-    def update_user_on_device(self, params):
-        tx = self.api.tm.transactions.transaction
-        with TransactionContextManager(tx) as api:
-            if self.is_version_less_than_13():
-                r = api.tm.auth.users.user.load(
-                    name=self.params['username_credential']
-                )
-                r.modify(**params)
+        when creating a user resource.
+        """
+        if self.want.password_credential is None:
+            err = "The 'password_credential' option " \
+                  "is required when creating a resource."
+            raise F5ModuleError(err)
+        if self.want.partition_access is None:
+            err = "The 'partition_access' option " \
+                  "is required when creating a resource."
+            raise F5ModuleError(err)
+
+    def validate_shell_parameter(self):
+        """Method to validate shell parameters.
+
+        Raise when 'none' is being set for shell parameter,
+        only if BIGIP does not return shell attribute in json.
+        This is when the attribute is set to none already.
+
+        Raise when shell attribute is set to 'bash' with roles set to
+        either 'admin' or 'resource-admin'.
+
+        NOTE: Admin and Resource-Admin roles automatically enable access to
+        all partitions, removing any other roles that the user might have
+        had. There are few other roles which do that but those roles,
+        do not allow bash.
+        """
+        if self.have is not None:
+            if self.want.shell == 'none' and self.have.shell is None:
+                error = "Attribute 'bash' is already set to 'none' " \
+                        "on target device."
+                raise F5ModuleError(error)
+
+        if self.want.shell == 'bash':
+            err = "Shell access is only available to 'admin' or " \
+                  "'resource-admin' roles"
+            permit = ['admin', 'resource-admin']
+            access_want = self.want.partition_access
+            # This can occur when we do checks at create
+            if self.have is not None:
+                access_have = self.have.partition_access
             else:
-                r = api.tm.auth.users.user.load(
-                    name=self.params['username_credential'],
-                    partition=self.params['partition']
-                )
-                r.modify(**params)
+                access_have = []
+            if access_want:
+                for access in access_want:
+                    if access['role'] not in permit:
+                        raise F5ModuleError(err)
+            if access_have:
```

Comment:
it was an example. just collapse it into an any expression

`x['role'] if x['role'] in permit for ....`

or whatever it becomes